### ISSUE TITLE
NamedParserRegistry matcher shouldn't be public

### ISSUE THREAD

**Issue by mibac138** [mibac138/ArgParser#16]
Why is `matcher` in `NamedParserRegistry` a public var? Doesn't this break encapsulation? If someone wants to use a different `ArgumentMatcher` it should be possible by passing it through the constructor.

**Comment by mibac138** [comment]
The problem is that some registries can be easily used in scenarios while others can't. This is noticeable by using a `SimpleParserRegistry` and a `MixedParserRegistry`. 
When using `SimplePR` you just add the `Parser`s you want to support to it and you are good to go. 
The situation with `MixedPR`s is different though. If you want to use them you need to explicitly specify which parser operates on which named argument or at which position can parse. In short `MixedPR`s needs resetup in case someone wants to use it in another scenario. 

_TLDR;_ the problem is that some `ParserRegistry`ies need resetup before using in another scenario whilst some don't.


This issue needs to be somehow addressed

**Comment by mibac138** [comment]
`matcher` is going to stay public (at least for now)
